packages/client/src/MessagePickupRepositoryClient.ts

```diff
@@ -94,6 +104,27 @@ export class MessagePickupRepositoryClient implements MessagePickupRepository {
     this.messagesReceivedCallback = callback
   }
 
+  /**
+   * Sets the callback function to retrieve the device token.
+   * This function is called whenever a token is needed, allowing dynamic retrieval based on the connection ID.
+   *
+   * @param callback - A function that receives a connectionId and returns a Promise resolving to a device token or undefined.
+   *
+   * @example
+   * // Example of setting the callback to retrieve the device token
+   * const client = new MessagePickupRepositoryClient({ url: 'wss://example.com' });
+   *
+   * const getDeviceToken = async (connectionId: string) => {
+   *   const connectionRecord = await agent.connections.findById(connectionId);
+   *   return connectionRecord?.getTag('device_token') as string | undefined;
+   * };
+   *
+   * client.notificationToken(getDeviceToken);
+   */
+  notificationToken(callback: (connectionId: string) => Promise<string | undefined>): void {
```

Comment:
I think we can make it more generic and call it `connectionInfoCallback`.

```suggestion
  setConnectionInfoCallback(callback: (connectionId: string) => Promise<ConnectionInfo | undefined>): void {
     this.connectionInfoCallback = callback
  }
```

Where:

```
export type ConnectionInfo = {
   fcmNotificationToken?: string
   maxReceiveBytes?: number
}
```

Reply:
Solved! 

In this update, I implemented a more generic callback function setConnectionInfo to allow dynamic retrieval of connection-specific information, such as the fcmNotificationToken and maxReceiveBytes. This callback replaces the previously token-specific callback, making it more versatile and adaptable for various connection-related data needs.

Key modifications:

- Introduced ConnectionInfo interface, which includes properties like fcmNotificationToken and maxReceiveBytes.

- Added setConnectionInfo method in MessagePickupRepositoryClient, allowing the client to set a callback that fetches ConnectionInfo based on connectionId.

- Updated addMessage and takeFromQueue methods to use the setConnectionInfoCallback for obtaining necessary connection data dynamically.

- Remove maxReceiveBytes definition of the constructor class MessagePickupRepositoryClient